src/hooks/useIosDevice.ts

```diff
@@ -0,0 +1,34 @@
+function detectIos() {
+  if (typeof window === 'undefined') return;
+
+  const isIosDevice = /iP(hone|od|ad)/.test(navigator.userAgent);
+  let versionOfIos = 0;
+
+  const versionMatch = navigator.userAgent.match(/OS (\d+)_(\d+)_?(\d+)?/);
+
+  if (versionMatch) {
+    versionOfIos = Number(`${versionMatch[1]}.${versionMatch[2]}`);
+
+    if (isNaN(versionOfIos)) {
+      /*
+        We need to ensure that we never return "NaN". It isn't referentially stable and
+        causes infinite loops inside logic like useEffect().
+      */
+      versionOfIos = 0;
+    }
+  }
+
+  return {
+    isIosDevice,
+    versionOfIos,
+  };
+}
+
+export function useIosDevice() {
```

Comment:
@calebjacob are these hooks, or would this be a better fit under `utils`?

Reply:
A hook is the right call I think since these values need to be relied on in other `useEffect()` and `useCallback()` calls.

Reply:
We could still have `useIosDevice()` but move `detectIos()` to the `utils` folder. Would you prefer that @charleslavon?

Reply:
No I don't feel strongly on that. Just trying to understand the differences in use/organization and make sure it's intentional

Reply:
Yes, let's move `detectIos` to the `utils` folder and make it exportable.

Reply:
But I'm also fine to go with this implementation

Reply:
Woops - sorry @shelegdmitriy, I didn't see your comment before I merged. Feel free to open a PR if you'd prefer to break out `detectIos()`.